sklearn/linear_model/_ridge.py

```diff
@@ -1616,6 +1616,17 @@ def fit(self, X, y, sample_weight=None):
             self.alpha_ = gs.best_estimator_.alpha
             self.best_score_ = gs.best_score_
 
+        min_alpha = min(self.alphas)
+        max_alpha = max(self.alphas)
+        if self.alpha_ in [min_alpha, max_alpha]:
```

Comment:
Here the test `test_ridge_cv_individual_penalties` is failing because the case `self.alpha_per_target=True` is tested. In that case `self.alpha_` is a vector, you need to find a way to check all the elements of the vector.